src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/Public.Data/ProcessorClientTests.cs

```diff
@@ -163,22 +163,6 @@ public async Task HttpClientBadRequest_ReturnsBadRequest()
             left.AssertValidationProblem(Errors.Error1);
         }
 
-        [Fact]
-        public async Task HttpClientNotFound_ReturnsNotFound()
-        {
-            var dataSetVersionId = Guid.NewGuid();
-
-            _mockHttp.Expect(HttpMethod.Delete, $"{Uri.AbsoluteUri}/{dataSetVersionId}")
-                .Respond(HttpStatusCode.NotFound);
-
-            var response = await _processorClient.DeleteDataSetVersion(dataSetVersionId);
-
-            _mockHttp.VerifyNoOutstandingExpectation();
-
-            var left = response.AssertLeft();
-            left.AssertNotFoundResult();
-        }
```

Comment:
I've reinstated this endpoint's special-case handling of 404s by allowing each endpoint to supply an optional "additionalResponseValidator" so that each endpoint can get first stab at deciding what to do with the response.  404s are now returning NotFoundResults for this endpoint alone, although I'd personally be tempted to consider doing this with all endpoints :)